src/components/Card/index.jsx

```diff
@@ -1,29 +1,88 @@
 import React, { Component } from 'react';
 import './index.css';
+import {BsPencil, BsX, BsCheck} from 'react-icons/bs';
 
 class Card extends Component {
     state = {
-        isChecked: false
+        isChecked: false,
+        isEdit: false,
+        cardData : {
+            Logo: 'Caption',
+            Text: 'Some text ...',
+        },
+        tempCardData: {
+            Logo: 'Caption',
+            Text: 'Some text ...'
+        }
     }
```

Comment:
I would say initial tempCardData should be empty. No need to store it from the beginning - since you should copy values once switching to Edit mode.